### PR TITLE
AzureResourceManagerTemplateDeployment fails authentication using a Bicep module registry.

### PR DESCRIPTION
**Pull request by v-schhabra**
**Task name**: AzureResourceManagerTemplateDeploymentV3

**Description**: Fixing ARM task

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) 

**Attached related issue:** (Y/N) https://mseng.visualstudio.com/DefaultCollection/AzureDevOps/_workitems/edit/2042790

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by ozanaydinMSFT — APPROVED** [review]

**Review by v-mohithgc — APPROVED** [review]

File: Tasks/AzureResourceManagerTemplateDeploymentV3/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -38,8 +38,6 @@
   "loc.input.help.deploymentOutputs": "Provide a name for the variable for the output variable which will contain the outputs section of the current deployment object in string format. You can use the “ConvertFrom-Json” PowerShell cmdlet to parse the JSON object and access the individual output values. For more details refer to [this](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/AzureResourceManagerTemplateDeploymentV3#deployment-outputs)",
   "loc.input.label.addSpnToEnvironment": "Access service principal details in override parameters",
   "loc.input.help.addSpnToEnvironment": "Adds service principal id and key of the Azure endpoint you chose to the script's execution environment. You can use these variables: `$servicePrincipalId` and `$servicePrincipalKey` in your override parameters like `-key $servicePrincipalKey`",
-  "loc.input.label.useWithoutJSON": "Use individual output values without JSON.Stringify applied",
```
Comment (ozanaydinMSFT) [inline Tasks/AzureResourceManagerTemplateDeploymentV3/Strings/resources.resjson/en-US/resources.resjson:41]: Please don't delete this lines and re-add.

File: Tasks/AzureResourceManagerTemplateDeploymentV3/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -38,8 +38,6 @@
...
-  "loc.input.help.useWithoutJSON": "Individual output values are being converted via JSON.Stringify by default. If you want to use the output values as it is without converting them via JSON.Stringify, enable this option. For more details refer to [this](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/AzureResourceManagerTemplateDeploymentV3#deployment-outputs)",
```
Comment (ozanaydinMSFT) [inline Tasks/AzureResourceManagerTemplateDeploymentV3/Strings/resources.resjson/en-US/resources.resjson:42]: Please don't delete this lines and re-add.

File: Tasks/AzureResourceManagerTemplateDeploymentV3/task.json [hunk 3]
```diff
@@ -225,15 +225,6 @@
             "required": false,
             "helpMarkDown": "Adds service principal id and key of the Azure endpoint you chose to the script's execution environment. You can use these variables: `$servicePrincipalId` and `$servicePrincipalKey` in your override parameters like `-key $servicePrincipalKey`",
             "groupName": "Advanced"
-        },
-        {
```
Comment (ozanaydinMSFT) [inline Tasks/AzureResourceManagerTemplateDeploymentV3/task.json:229]: Please don't delete this lines and re-add.

File: Tasks/AzureResourceManagerTemplateDeploymentV3/task.loc.json [hunk 4]
```diff
@@ -225,15 +225,6 @@
       "required": false,
       "helpMarkDown": "ms-resource:loc.input.help.addSpnToEnvironment",
       "groupName": "Advanced"
-    },
-    {
```
Comment (ozanaydinMSFT) [inline Tasks/AzureResourceManagerTemplateDeploymentV3/task.loc.json:229]: Please don't delete this lines and re-add.

**Pull request merged**